### PR TITLE
Convert Z to list if Z is str type on is_valid_backdoor_adjustment_set.

### LINKED ISSUE #1511: Z on is_valid_backdoor_adjustment_set does not accept str type.
### Subject of the issue
When Z on is_valid_backdoor_adjustment_set is str type, it fails and throw ValueError exception.

### Your environment
* pgmpy version 0.1.17
* Python version 3.9.6
* Operating System: macOS 12.2.1

### Steps to reproduce
Tell us how to reproduce this issue. Please provide a **minimal reproducible code** of
the issue you are facing if possible.

```
from pgmpy.inference import CausalInference
from pgmpy.models import BayesianModel

model = BayesianModel(
    [('abc', 'def'),('ghi', 'abc'),('ghi', 'def')])
causal_model = CausalInference(model)
causal_model.is_valid_backdoor_adjustment_set(X="abc",Y="ghi", Z="def")
```

### Expected behaviour
Tell us what should happen

It returns valid response `True` or `False`.

### Actual behaviour
Tell us what happens instead

It throw `ValueError: Node d not in not in graph`

### PR DESCRIPTION
**Pull request by 1betatsu**
### Your checklist for this pull request
Please review the [guidelines for contributing](CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your master!
- [x] Make sure you are making a pull request against the **dev branch** (left side). Also you should start *your branch* off *our dev*.
- [x] Check the commit's or even all commits' message styles matches our requested structure.

### Issue number(s) that this pull request fixes
- Fixes #1511 

### List of changes to the codebase in this pull request
- Add a logic that checks a type of Z and converts Z to a list if Z is str. This prevent Z of str from being separated into a list of single characters by `Z_ = list(Z)`.

### CODE REVIEW

File: pgmpy/inference/CausalInference.py [hunk 1]
```diff
@@ -97,6 +97,8 @@ def is_valid_backdoor_adjustment_set(self, X, Y, Z=[]):
         >>> inference.is_valid_backdoor_adjustment_set("X", "Y")
         True
         """
+        if isinstance(Z, str):
+            Z = [Z]
         Z_ = list(Z)
```
Comment (ankurankan) [inline pgmpy/inference/CausalInference.py]: @1betatsu Thanks for the PR. Could possibly make this a bit cleaner by replacing this line with something like: `Z_ = [Z] if isinstance(Z, str) else list(Z))`. Also could you please add a test for this change?

Comment (1betatsu) [reply pgmpy/inference/CausalInference.py]: @ankurankan Thank you for your quick feedback!
I would add tests to validate Z of is_valid_backdoor_adjustment_set. I also forgot formatting by black. I would fix it, too.

**Comment by 1betatsu** [comment]
@ankurankan I finally passed all checks.

Sorry for redundant commits due to my lack of experience in black formatting.

**Comment by ankurankan** [comment]
@1betatsu No worries, thanks a lot for the fixes.

**Pull request merged**